lib/internal/streams/readable.js

```diff
@@ -1154,6 +1155,39 @@ async function* createAsyncIterator(stream, options) {
   }
 }
 
+function staticUnref(stream) {
+  // TODO - We can probably have a better way to do this
+  function unref() {
+    return async function* unref() {
+      const iterator = this.iterator({ destroyOnReturn: false });
+
+      for await (const val of iterator) {
+        yield val;
+      }
+    }.call(this);
+  }
+
+  // Instead of regular operator registration
+  const newStream = Readable.from(ReflectApply(unref, stream, []));
+
+  // TODO - we should have better way instead of listening to those events, no?
+  // stream.once('close', () => {
+  //   destroyImpl.destroyer(newStream, null);
+  // });
```

Comment:
uncommenting this fixes the `original stream close should close unref one` test but fails the `Should close original stream when unref one consume all data` test because the `newStream` is not finished yet for some reason so it will get an abort error